### PR TITLE
Readability improvements

### PR DESCRIPTION
**Pull request by KGKallasmaa**
Making the excising code more readable

### CODE REVIEW

File: alibi/confidence/model_linearity.py [hunk 1]
```diff
@@ -65,14 +65,12 @@ def _calculate_global_linearity(predict_fn: Callable, input_shape: Tuple, X_samp
     elif model_type == 'regressor':
         outs = predict_fn(X_samples)
         outs_shape = outs.shape[1:]
-        if len(outs.shape) == 1:
-            outs = outs.reshape(ss + (1,))  # shape=(nb_instances, nb_samples, 1)
-        else:  # if regression on multiple targets
-            outs = outs.reshape(ss + outs_shape)  # shape=(nb_instances, nb_samples, nb_targets)
+        outs = outs.reshape(ss + (1,)) if len(outs.shape) == 1 else outs.reshape(ss + outs_shape)
```
Comment (ascillitoe) [inline alibi/confidence/model_linearity.py:68]: Would be good to retain comments here.

File: alibi/confidence/model_linearity.py [hunk 2]
```diff
@@ -65,14 +65,12 @@ def _calculate_global_linearity(predict_fn: Callable, input_shape: Tuple, X_samp
...
+
     else:
         raise ValueError("Passed 'model_type' not supported. Supported model types: 'classifier', 'regressor'")
     t_f = time() - t_0
-    logger.debug('predict time {}'.format(t_f))
+    logger.debug(f'predict time {t_f}')
```
Comment (ascillitoe) [inline alibi/confidence/model_linearity.py:73]: Historically we have the convention of not using f-strings with `logger`. See [here](https://klichx.dev/2019/11/09/f-strings-and-python-logging/).

File: alibi/confidence/model_linearity.py [hunk 3]
```diff
@@ -84,23 +82,19 @@ def _calculate_global_linearity(predict_fn: Callable, input_shape: Tuple, X_samp
...
-        out_sum_shape = out_sum.shape[1:]
     elif model_type == 'regressor':
         out_sum = predict_fn(summ)
         out_sum_shape = out_sum.shape[1:]
-        if len(out_sum.shape) == 1:
-            out_sum = out_sum.reshape((ss[0],) + (1,))
-        else:
-            out_sum = out_sum.reshape((ss[0],) + out_sum_shape)
+        out_sum = out_sum.reshape((ss[0],) + (1,)) if len(out_sum.shape) == 1 else out_sum.reshape(
+            (ss[0],) + out_sum_shape)
```
Comment (ascillitoe) [inline alibi/confidence/model_linearity.py:89]: Not 100% convinced this one is actually more readable, but it is quite subjective I guess...

File: alibi/datasets/default.py [hunk 4]
```diff
@@ -213,8 +213,7 @@ def fetch_movie_sentiment(return_X_y: bool = False, url_id: int = 0) -> Union[Bu
     return Bunch(data=data, target=labels, target_names=target_names)
 
 
-def fetch_adult(features_drop: Optional[list] = None, return_X_y: bool = False, url_id: int = 0) -> \
-        Union[Bunch, Tuple[np.ndarray, np.ndarray]]:
+def fetch_adult(features_drop: Optional[list] = None, return_X_y: bool = False, url_id: int = 0) -> Union[Bunch, Tuple[np.ndarray, np.ndarray]]:
```
Comment (ascillitoe) [inline alibi/datasets/default.py:216]: Surely this line is too long?

File: alibi/datasets/default.py [hunk 5]
```diff
@@ -325,9 +324,9 @@ def fetch_adult(features_drop: Optional[list] = None, return_X_y: bool = False,
 
     # only return data values
     data = data.values
-    target_names = ['<=50K', '>50K']
-
     if return_X_y:
         return data, labels
 
+    target_names = ['<=50K', '>50K']
+
```
Comment (ascillitoe) [inline alibi/datasets/default.py:331]: What's the logic behind moving this here? It makes more sense defining `target_names` next to `data` to me...

File: alibi/confidence/model_linearity.py (hunk 2 above)
Comment (jklaise) [reply alibi/confidence/model_linearity.py:73]: I believe the logging statements should be written with string interpolation `%s` syntax.

File: alibi/datasets/tensorflow.py [hunk 6]
```diff
@@ -23,14 +23,14 @@ def fetch_fashion_mnist(return_X_y: bool = False
...
-        5: 'Sandal', 6: 'Shirt', 7: 'Sneaker', 8: 'Bag', 9: 'Ankle boot',
-    }
-
     data, labels = keras.datasets.fashion_mnist.load_data()[0]
 
     if return_X_y:
         return data, labels
 
+    target_names = {
+        0: 'T-shirt/top', 1: 'Trouser', 2: 'Pullover', 3: 'Dress', 4: 'Coat',
```
Comment (ascillitoe) [inline alibi/datasets/tensorflow.py:32]: Again, I think its better to keep `target_names` defined next to `data` and `labels`...

File: alibi/explainers/ale.py [hunk 7]
```diff
@@ -276,8 +276,7 @@ def get_quantiles(values: np.ndarray, num_quantiles: int = 11, interpolation='li
 
     """
     percentiles = np.linspace(0, 100, num=num_quantiles)
-    quantiles = np.percentile(values, percentiles, axis=0, interpolation=interpolation)  # type: ignore[call-overload]
-    return quantiles
+    return np.percentile(values, percentiles, axis=0, interpolation=interpolation)
```
Comment (ascillitoe) [inline alibi/explainers/ale.py:279]: Is the `# type: ignore[call-overload]` not needed anymore?

File: alibi/datasets/default.py (hunk 5 above)
Comment (jklaise) [reply alibi/datasets/default.py:331]: I believe it's so that we don't define it needlessly if the function returns early, though it's a minor point.

**Comment by ascillitoe** [comment]
Hi @KGKallasmaa thanks for this. I started by making a few comments, but would suggest resolving the `flake8` and possibly `mypy` errors before we go any further. FYI you can run these locally (see [here](https://github.com/SeldonIO/alibi/blob/master/CONTRIBUTING.md#linter)) once you've run `pip install -r requirements/dev.txt`.

**Comment by jklaise** [comment]
Will close this as the code has diverged somewhat...
